tests/com.espressif.idf.ui.test/src/com/espressif/idf/ui/test/executable/cases/project/NewEspressifIDFProjectSBOMTest.java

```diff
@@ -0,0 +1,216 @@
+		{
+			SWTBotButton okButton = bot.button("OK");
+			assertTrue(!okButton.isEnabled());
+			bot.checkBox("Redirect output to the file").click();
+			assertTrue(!okButton.isEnabled());
+			bot.checkBox("Redirect output to the file").click();
+			assertTrue(!okButton.isEnabled());
+			bot.button("Cancel").click();
+		}
+
+		private static void whenRunSBOMtool() throws IOException
+		{
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "SBOM Tool");
+			bot.button("OK").click();
+			ProjectTestOperations
+					.joinJobByName(com.espressif.idf.ui.update.Messages.SbomCommandDialog_EspIdfSbomJobName);
+		}
+
+		private static void thenCheckResultInConsole() throws IOException
+		{
+			ProjectTestOperations.findInConsole(bot, "Espressif IDF Tools Console", "SPDXVersion");
+		}
+
+		private static void thenVerifyConsole() throws IOException
+		{
+			ProjectTestOperations.findInConsole(bot, "Espressif IDF Tools Console",
+					"The output was redirected to the file:");
+		}
+
+		private static void whenRunSBOMtoolUsingContextMenuRedirectOutputToFile() throws IOException
+		{
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "SBOM Tool");
+			bot.checkBox("Redirect output to the file").click();
+			bot.button("OK").click();
+			ProjectTestOperations
+					.joinJobByName(com.espressif.idf.ui.update.Messages.SbomCommandDialog_EspIdfSbomJobName);
+		}
+
+		private static void thenCheckResultInFile() throws IOException
+		{
+			ProjectTestOperations.findInConsole(bot, "Espressif IDF Tools Console",
+					"The output was redirected to the file:");
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "Refresh");
+			bot.tree().getTreeItem(projectName).getNode("sbom.txt").select();
+			bot.tree().getTreeItem(projectName).getNode("sbom.txt").doubleClick();
+			assertTrue(ProjectTestOperations.checkTextEditorContentForPhrase("SPDXVersion", bot));
+		}
+
+		private static void thenCheckPathValidation(SWTWorkbenchBot bot) throws IOException
+		{
+			SWTBotButton okButton = bot.button("OK");
+			bot.checkBox("Redirect output to the file").click();
+			if (!SystemUtils.IS_OS_LINUX)
+			{
+				ProjectTestOperations.setTextFieldInShell(bot, "Software Bill of Materials Tool", "Output File Path",
+						" ");
+				assertTrue(!okButton.isEnabled());
+			}
+			ProjectTestOperations.setTextFieldInShell(bot, "Software Bill of Materials Tool", "Output File Path", "");
+			ProjectTestOperations.setTextFieldInShell(bot, "Software Bill of Materials Tool",
+					"Project Description Path", "");
+			assertTrue(!okButton.isEnabled());
+			ProjectTestOperations.setTextFieldInShell(bot, "Software Bill of Materials Tool",
+					"Project Description Path", " ");
+			assertTrue(!okButton.isEnabled());
+			ProjectTestOperations.setTextFieldInShell(bot, "Software Bill of Materials Tool",
+					"Project Description Path", "\\");
+			assertTrue(!okButton.isEnabled());
+			bot.button("Cancel").click();
+		}
+	}
```

Comment:
This whole function needs reword. Then should  be only verifying some ui elements or other items if needed it should not be carrying any operations. all the setTextField items should be moved in when method.